From 07401030151c7f7c0f9b57b51011b0ac2a476040 Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Fri, 22 Jun 2018 13:31:56 +0800
Subject: [PATCH 4514/5242] MLK-19413-3 arm64: fsl-imx8dx.dtsi: Add pixel
 combiner node

commit  da7ac9c39dbea808f2bb92e99efb5925d1a9bf16 from
https://source.codeaurora.org/external/imx/linux-imx.git

This patch adds pixel combiner node support for i.MX8dx DT
file and hooks the pixel combiner node to the DPU node.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi |    8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
index f4f905f..dae0b08 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8dx.dtsi
@@ -1481,6 +1481,13 @@
 		reg = <0x0 0x56000000 0x0 0x10000>;
 	};
 
+	pixel_combiner: pixel-combiner@56020000 {
+		compatible = "fsl,imx8qxp-pixel-combiner";
+		reg = <0x0 0x56020000 0x0 0x10000>;
+		power-domains = <&pd_dc0>;
+		status = "disabled";
+	};
+
 	prg1: prg@56040000 {
 		compatible = "fsl,imx8qxp-prg", "fsl,imx8qm-prg";
 		reg = <0x0 0x56040000 0x0 0x10000>;
@@ -1690,6 +1697,7 @@
 		fsl,dpr-channels = <&dpr1_channel1>, <&dpr1_channel2>,
 				   <&dpr1_channel3>, <&dpr2_channel1>,
 				   <&dpr2_channel2>, <&dpr2_channel3>;
+		fsl,pixel-combiner = <&pixel_combiner>;
 		status = "disabled";
 
 		dpu_disp0: port@0 {
-- 
1.7.9.5

